site/content/api/Event-API.md

```diff
@@ -76,4 +81,82 @@ The following sequence diagram illustrates the flow of messages involved in a *B
 
 **Message Format**
 
+**Arbitrary events from the device**
+
 See [*Telemetry API*]({{< relref "Telemetry-API.md" >}}) for definition of message format. 
+
+## Well-known event message types
+
+Hono defines *well-known* events that are of a specific *content-type*.
+
+NB: currently there is only one such *well-known* event.
+
+### Lifecycle Notifications
+
+A specific event can signal that a device is `connected` (e.g. to signal that it is ready to receive a command for a specific time interval) 
+or that a device is `disconnected` (e.g to signal that it is not ready to receive a command anymore).
+ 
+*Business Applications* may want to react to such an event by sending a command upstream to the device.
```

Comment:
FMPOV this is **not** the purpose of this event. We cannot prevent applications from trying to send a command on reception of this event, but the real trigger for sending a command should be another event, specifically designed for that purpose.

Reply:
No, we cannot prevent that, but IMHO we had *not* agreed on two events yet.

What I did instead was to define optional parts into this event: the device can state how long it tries to stay connected and specify by that it can make sense to send a command.

We can strictly separate it:
- connected/disconnected *only* from protocol adapters (often this might not make sense, see piggy-backing telemetry data)
- command readiness only from device itself

Not convinced much about the first event though - we wanted to prevent the doubling of messages in case of telemetry messages for devices connecting via HTTP e.g.
Can imagine that protocol adapters can be configured for a tenant to always issue the first event type (for certain device setups).

Will include both events in the next commit.

Reply:
> connected/disconnected only from protocol adapters (often this might not make sense, see piggy-backing telemetry data)

This is exactly the use-case that the Kapua folks seem to have in mind. For Kapua, it is helpful to have an indicator of the connection status of a gateway. In the Kapua case we are only talking MQTT based gateways (Kura) which connect on startup and stay connected all the time. So this event is basically only used for feeding a *gateway connected* LED in the UI.

As such, I think it makes a lot of sense to

* have separate events for the two distinct purposes and
* make the emitting of the connected/disconnected events configurable per tenant and adapter.

Reply:
Ok, this makes sense to me too: we use the tenant based configuration of protocol adapters to let them emit the connected/disconnected event (if configured).

So I will introduce both events in next commits (but will for now do some minor prototyping first).